WordPress/build.gradle

```
@@ -511,12 +535,18 @@ tasks.register("updateVersionProperties") {
 
 tasks.register("printVersionName") {
     doLast {
-        def key = "$productKey" + "."
-        if(project.hasProperty('alpha')) {
-            key += "alpha."
+        def app = project.findProperty('productKey') ?: 'wordpress'
+        def buildType = project.findProperty('buildType') ?: ''
+        println versionForVariant(app, buildType)['versionName']
+    }
+}
+
+tasks.register("printAllVersions") {
+    doLast {
+        android.applicationVariants.all { variant ->
+            def apkData = variant.outputs[0].apkData
+            println "${variant.name}: ${apkData.versionName} (${apkData.versionCode})"
```

I wonder if we should iterate through all the outputs just to cover future changes 🤔 

Maybe we can do something like this:

```suggestion
            def apkData = variant.outputs.collect { it.apkData }
            println "${variant.name}: ${apkData.collect { it.versionName }} ${apkData.collect { it.versionCode }}"
```

This will put the versionName and versionCode between `[]` since they'll be lists, but that probably doesn't matter too much since this is essentially a debug task.

What do you think @AliSoftware ?

---

Works for me. I actually discovered a nice syntax while browsing for Groovy syntax for `.map` / `.collect` yesterday, it seems `*.` is a shorthand for map/collect. Just tried it and seems to work well, while keeping the code compact and really nice to read:

```suggestion
            def apkData = variant.outputs*.apkData
            println "${variant.name}: ${apkData*.versionName} (${apkData*.versionCode})"
```

---

Actually directly pushed this change on `release/17.6` just now to not forget about it (was trivial enough that I figured it was not worth a PR): 441604e104

---

Fwiw, I am not a big fan of merging any change, however trivial it might be, without giving a chance for the reviewer to check it out. I understand why you preferred to do it this way and I've also done this on a few occasions, so all good 👍